src/plugins/index_pattern_editor/public/components/empty_prompts/empty_prompts.tsx

```diff
@@ -17,32 +18,44 @@ import { getIndices } from '../../lib';
 import { EmptyIndexListPrompt } from './empty_index_list_prompt';
 import { EmptyIndexPatternPrompt } from './empty_index_pattern_prompt';
 import { PromptFooter } from './prompt_footer';
+import { KNOWN_FLEET_ASSETS } from '../../../../data/common';
 
 const removeAliases = (item: MatchedItem) =>
   !((item as unknown) as ResolveIndexResponseItemAlias).indices;
 
 interface Props {
   onCancel: () => void;
   allSources: MatchedItem[];
-  hasExistingIndexPatterns: boolean;
   loadSources: () => void;
 }
 
-export const EmptyPrompts: FC<Props> = ({
-  hasExistingIndexPatterns,
-  allSources,
-  onCancel,
-  children,
-  loadSources,
-}) => {
+export function isDataIndex(source: MatchedItem) {
```

Comment:
I am less than happy with this because it adds complexity by spreading this logic between different places. It would be a lot nicer if we could get away with containing everything in a single REST API, 

I wonder if handling this is actually such an edge case that we can ignore this? Then we could simplify this pr quite a bit. 

This affects the following scenario: 

When a user navigates to the index pattern page with no data or index patterns, we open a flyout with an empty state. The empty state could be in two states:
1. No data
2. No index patterns

then:

* If to remove this code, then existing `metrics-elastic_agent` and `logs-elastic_agent` would be considered as user data and we will show a "No index pattern" state. 
* If to leave this code, then we will show a "No data" state

Reply:
Also welcome other ideas